scheduler/core/sources.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+# For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+from enum import Enum
+from abc import ABC, abstractmethod
+from typing import ClassVar, Optional, NoReturn, Tuple
+from scheduler.services.resource import OcsResourceService, FileResourceService
+from scheduler.core.resourcemanager import ExternalService
+from scheduler.services.environment import Env
+from io import BytesIO
+
+class Services(Enum):
+    ENV = 'env'
+    RESOURCE = 'resource'
+    CHRONICLE = 'chronicle'
+
+class Origin(ABC):
+
+
+    def __init__(self,
+                 resource: Optional[ExternalService] = None,
+                 env: Optional[ExternalService] = None,
+                 chronicle: Optional[ExternalService] = None,
+                 is_loaded: bool = False):
+        self.resource = resource
+        self.env = env
+        self.chronicle = chronicle
+        self.is_loaded = is_loaded
+
+    @abstractmethod
+    def load(self) -> NoReturn:
```

Comment:
This should be `None`, but we can handle that in one big PR since we have to change it all over the place.